pkg/sampler/sampler.go

```diff
@@ -33,15 +35,13 @@ func (t *Target) SetHash() {
 
 type Sample struct {
 	StatusCode      int
-	T1              time.Time
-	T2              time.Time
-	ResponseHeaders map[string]string
-}
-
-// Latency returns the amount of milliseconds between T1
-// and T2 (start and finish).
-func (s Sample) Latency() float64 {
-	return s.T2.Sub(s.T1).Seconds() * 1000
+	TimeStart       time.Time
+	TimeToConnect   time.Time
+	TimeToFirstByte time.Time
+	TimeEnd         time.Time
```

Comment:
I like the idea of these being bonafide timestamps first, and then allowing downstream things to determine deltas.

But... if I'm going to do that, I should probably rename them to `TimeConnect` and `TimeFirstByte`.